cpp/include/cudf/detail/rolling/rolling.cuh

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright (c) 2019-2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
```

Comment:
@harrism, please forgive my having refactored this out of `rolling.cu`, and into a `detail` header. I'm afraid this was required for the `groupby::windowed_aggregate()`. I realize this might have impact on the Python-UDF-based windowing that you're working on.